test/e2e_federation/replicaset.go

```diff
@@ -392,6 +392,9 @@ func updateReplicaSetOrFail(clientset *fedclientset.Clientset, replicaset *v1bet
 	}
 	By(fmt.Sprintf("Updating federation replicaset %q in namespace %q", replicaset.Name, namespace))
 
+	// a hack to pass validation for controller selector immutability prior update
+	replicaset.Spec.Selector.MatchLabels = map[string]string{"name": "myrs-consistent"}
```

Comment:
This hack is needed as a new selector is generated every time creating a new RS for updating purpose. Refer [this](https://github.com/kubernetes/kubernetes/blob/master/test/e2e_federation/replicaset.go#L403-L407) for more information.

Aside from this, there are two ways to resolve this:
(1) make the affected tests run in serial mode (drawback: slow)
(2) create a constant and pass it as function parameter